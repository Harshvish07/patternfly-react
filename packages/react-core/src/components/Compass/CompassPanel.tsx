import styles from '@patternfly/react-styles/css/components/Compass/compass';
import { css } from '@patternfly/react-styles';

interface CompassPanelProps extends React.HTMLProps<HTMLDivElement> {
  /** Content of the panel. */
  children: React.ReactNode;
  /** Additional classes added to the panel. */
  className?: string;
  /** Indicates the panel should have a pill border radius */
  isPill?: boolean;
  /** Indicates the panel should expand to fill the available height */
  isFullHeight?: boolean;
  /** Indicates the panel should scroll its overflow */
  isScrollable?: boolean;
  /** Indicates the panel should have no border */
  hasNoBorder?: boolean;
  /** Indicates the panel should have no padding */
  hasNoPadding?: boolean;
  /** Indicates the panel should have a "thinking" animation */
  isThinking?: boolean;
}

export const CompassPanel: React.FunctionComponent<CompassPanelProps> = ({
  children,
  className,
  isPill,
  hasNoBorder,
  hasNoPadding,
  isThinking,
  isFullHeight,
  isScrollable,
  ...props
}) => (
  <div
    className={css(
      styles.compassPanel,
      isPill && styles.modifiers.pill,
      hasNoBorder && styles.modifiers.noBorder,
      hasNoPadding && styles.modifiers.noPadding,
      isThinking && 'pf-v6-m-thinking',
      isFullHeight && styles.modifiers.fullHeight,
      isScrollable && styles.modifiers.scrollable,
      className
    )}
    {...props}
  >
    {children}
  </div>
);

CompassPanel.displayName = 'CompassPanel';
