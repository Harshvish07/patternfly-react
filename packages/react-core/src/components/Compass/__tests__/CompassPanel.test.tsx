import { render, screen } from '@testing-library/react';
import { CompassPanel } from '../CompassPanel';
import styles from '@patternfly/react-styles/css/components/Compass/compass';

test('Renders with children', () => {
  render(<CompassPanel>Test content</CompassPanel>);
  expect(screen.getByText('Test content')).toBeVisible();
});

test('Renders with custom class name when className prop is provided', () => {
  render(<CompassPanel className="custom-class">Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass('custom-class');
});

test(`Renders with default ${styles.compassPanel} class`, () => {
  render(<CompassPanel>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.compassPanel);
});

test(`Renders with ${styles.modifiers.pill} when isPill is true`, () => {
  render(<CompassPanel isPill>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.modifiers.pill);
});

test(`Renders with ${styles.modifiers.noBorder} when hasNoBorder is true`, () => {
  render(<CompassPanel hasNoBorder>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.modifiers.noBorder);
});

test(`Renders with ${styles.modifiers.noPadding} when hasNoPadding is true`, () => {
  render(<CompassPanel hasNoPadding>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.modifiers.noPadding);
});

test('Renders with pf-v6-m-thinking when isThinking is true', () => {
  render(<CompassPanel isThinking>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass('pf-v6-m-thinking');
});

test(`Renders with ${styles.modifiers.fullHeight} when isFullHeight is true`, () => {
  render(<CompassPanel isFullHeight>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.modifiers.fullHeight);
});

test(`Renders with ${styles.modifiers.scrollable} when isScrollable is true`, () => {
  render(<CompassPanel isScrollable>Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveClass(styles.modifiers.scrollable);
});

test('Renders with multiple modifier classes when multiple props are true', () => {
  render(
    <CompassPanel isPill hasNoBorder hasNoPadding isThinking isFullHeight isScrollable>
      Test
    </CompassPanel>
  );
  const panelElement = screen.getByText('Test');
  expect(panelElement).toHaveClass(styles.modifiers.pill);
  expect(panelElement).toHaveClass(styles.modifiers.noBorder);
  expect(panelElement).toHaveClass(styles.modifiers.noPadding);
  expect(panelElement).toHaveClass('pf-v6-m-thinking');
  expect(panelElement).toHaveClass(styles.modifiers.fullHeight);
  expect(panelElement).toHaveClass(styles.modifiers.scrollable);
});

test('Renders with additional props spread to the component', () => {
  render(<CompassPanel aria-label="Test label">Test</CompassPanel>);
  expect(screen.getByText('Test')).toHaveAccessibleName('Test label');
});

test('Matches the snapshot with all modifiers', () => {
  const { asFragment } = render(
    <CompassPanel isPill hasNoBorder hasNoPadding isThinking isFullHeight isScrollable>
      <div>Panel with all modifiers</div>
    </CompassPanel>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Matches the snapshot with no modifiers', () => {
  const { asFragment } = render(
    <CompassPanel>
      <div>Basic panel</div>
    </CompassPanel>
  );
  expect(asFragment()).toMatchSnapshot();
});
