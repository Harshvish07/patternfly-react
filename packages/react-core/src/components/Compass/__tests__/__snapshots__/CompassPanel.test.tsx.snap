// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-compass__panel pf-m-pill pf-m-no-border pf-m-no-padding pf-v6-m-thinking pf-m-full-height pf-m-scrollable"
  >
    <div>
      Panel with all modifiers
    </div>
  </div>
</DocumentFragment>
`;

exports[`Matches the snapshot with no modifiers 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-compass__panel"
  >
    <div>
      Basic panel
    </div>
  </div>
</DocumentFragment>
`;
